### PR TITLE
Fix expiration watcher comparator

### PR DESCRIPTION
**Pull request by LogvinovLeon**
## Description

RB tree is unable to store values with the same key. Before - we user expiraration time as a key.
This PR changes that and adds order hash as a secondary key

## How Has This Been Tested?

Added a regression test

## Types of changes

*   [ ] Bug fix (non-breaking change which fixes an issue)
*   [ ] New feature (non-breaking change which adds functionality)
*   [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

*   [ ] Change requires a change to the documentation.
*   [ ] Added tests to cover my changes.
*   [ ] Added new entries to the relevant CHANGELOGs.
*   [ ] Updated the new versions of the changed packages in the relevant CHANGELOGs.
*   [ ] Labeled this PR with the 'WIP' label if it is a work in progress.
*   [ ] Labeled this PR with the labels corresponding to the changed package.

### CODE REVIEW

File: packages/0x.js/src/order_watcher/expiration_watcher.ts [hunk 1]
```diff
@@ -23,7 +23,15 @@ export class ExpirationWatcher {
         this._orderExpirationCheckingIntervalMs =
             expirationMarginIfExistsMs || DEFAULT_ORDER_EXPIRATION_CHECKING_INTERVAL_MS;
         const scoreFunction = (orderHash: string) => this._expiration[orderHash].toNumber();
-        const comparator = (lhs: string, rhs: string) => scoreFunction(lhs) - scoreFunction(rhs);
+        const comparator = (lhsOrderHash: string, rhsOrderHash: string) => {
+            const lhsExpiration = this._expiration[lhsOrderHash].toNumber();
```
Comment (BMillman19) [inline packages/0x.js/src/order_watcher/expiration_watcher.ts]: are we still using `scoreFunction` anywhere? should we be calling that here?

Comment (LogvinovLeon) [reply packages/0x.js/src/order_watcher/expiration_watcher.ts]: No. Good catch

File: packages/migrations/package.json [hunk 2]
```diff
@@ -13,7 +13,7 @@
         "migrate": "run-s build compile script:migrate",
         "script:migrate": "node ./lib/migrate.js",
         "copy_artifacts": "copyfiles 'src/artifacts/**/*' ./lib",
-        "compile": "node ../deployer/lib/src/cli.js compile --contracts ${npm_package_config_contracts} --contracts-dir ../contracts/src/contracts --artifacts-dir src/artifacts"
+        "compile": "node ../deployer/lib/src/cli.js compile --contracts ${npm_package_config_contracts} --contract-dirs ../contracts/src/contracts --artifacts-dir src/artifacts"
```
Comment (fabioberger) [inline packages/migrations/package.json]: Is this a merge conflict?

Comment (LogvinovLeon) [reply packages/migrations/package.json]: Yes

File: packages/0x.js/test/global_hooks.ts [hunk 3]
```diff
@@ -0,0 +1,10 @@
+// HACK: This dependency is optional since it is only available when run from within
+// the monorepo. tslint doesn't handle optional dependencies
+// tslint:disable-next-line:no-implicit-dependencies
```
Comment (fabioberger) [inline packages/0x.js/test/global_hooks.ts]: Remove this comment since it's no longer true.

**Pull request merged**